sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRepositoryClientBuilder.java

```diff
@@ -0,0 +1,295 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+package com.azure.containers.containerregistry;
+
+import com.azure.containers.containerregistry.implementation.authentication.ContainerRegistryCredentialsPolicy;
+import com.azure.containers.containerregistry.models.ContainerRegistryServiceVersion;
+import com.azure.core.annotation.ServiceClientBuilder;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.HttpHeader;
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpPipelineBuilder;
+import com.azure.core.http.HttpPipelinePosition;
+import com.azure.core.http.policy.AddDatePolicy;
+import com.azure.core.http.policy.AddHeadersPolicy;
+import com.azure.core.http.policy.CookiePolicy;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.core.http.policy.HttpLoggingPolicy;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.http.policy.HttpPolicyProviders;
+import com.azure.core.http.policy.RequestIdPolicy;
+import com.azure.core.http.policy.RetryPolicy;
+import com.azure.core.http.policy.UserAgentPolicy;
+import com.azure.core.util.ClientOptions;
+import com.azure.core.util.Configuration;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.serializer.JacksonAdapter;
+import com.azure.core.util.serializer.SerializerAdapter;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/** A builder for creating a new instance of the ContainerRegistry type. */
+@ServiceClientBuilder(
+    serviceClients = {
+        ContainerRepositoryClient.class,
+        ContainerRepositoryAsyncClient.class
+    })
+public final class ContainerRepositoryClientBuilder {
+    private static final Map<String, String> PROPERTIES =
+        CoreUtils.getProperties("azure-containers-containerregistry.properties");
+
+    private static final String CLIENT_NAME = PROPERTIES.getOrDefault("name", "UnknownName");
+    private static final String CLIENT_VERSION = PROPERTIES.getOrDefault("version", "UnknownVersion");
+
+    private final ClientLogger logger = new ClientLogger(ContainerRepositoryClientBuilder.class);
+    private final List<HttpPipelinePolicy> perCallPolicies = new ArrayList<>();
+    private final List<HttpPipelinePolicy> perRetryPolicies = new ArrayList<>();
+    private ClientOptions clientOptions;
+    private Configuration configuration;
+    private String endpoint;
+    private HttpClient httpClient;
+    private TokenCredential credential;
+    private HttpPipeline httpPipeline;
+    private HttpLogOptions httpLogOptions;
+    private SerializerAdapter serializerAdapter;
```

Comment:
serializerAdapter can be removed (probably should be removed until Container Registry has any requirements on custom serialization handling)